backend/moonstreamapi/admin/migrations/generate_entity_subscriptions.py

```diff
@@ -0,0 +1,651 @@
+                    try:
+                        subscription_resource: BugoutResource = bc.create_resource(
+                            token=MOONSTREAM_ADMIN_ACCESS_TOKEN,
+                            application_id=MOONSTREAM_APPLICATION_ID,
+                            resource_data=resource_data,
+                        )
+                        stages[user_id]["subscription_resource_id"] = str(
+                            subscription_resource.id
+                        )
+                    except Exception as e:
+                        logger.error(
+                            f"Failed to create subscription brood resource: {str(e)}"
+                        )
+
+            if "proccessed_subscriptions" not in stages[user_id]:
+                stages[user_id]["proccessed_subscriptions"] = {}
+
+            ### Add subscriptions to collection
+
+            for subscription in subscriptions:
+                if (
+                    str(subscription["subscription_id"])
+                    in stages[user_id]["proccessed_subscriptions"]
+                ):
+                    continue
+
+                subscription_type_id = subscription["subscription_type_id"]
+                address = subscription["address"]
+                color = subscription["color"]
+                label = subscription["label"]
+
+                # try to get abi from S3
+                abi = None
+                if subscription.get("bucket") and subscription.get("s3_path"):
+                    try:
+                        abi_body = get_abi_from_s3(
+                            bucket=subscription["bucket"],
+                            s3_path=subscription["s3_path"],
+                        )
+                        # abi hash
+                        abi_hash = hashlib.sha256(abi_body.encode("utf-8")).hexdigest()
+                        abi = True
+                    except Exception as e:
+                        logger.error(f"Failed to get abi from S3: {str(e)}")
+                        abi = None
+
+                entity = add_entity_subscription(
+                    user_id=user_id,
+                    collection_id=collection_id,
+                    subscription_type_id=subscription_type_id,
+                    address=address,
+                    color=color,
+                    label=label,
+                    content={"abi": abi_body, "abi_hash": abi_hash} if abi else {},
+                )
+                stages[user_id]["proccessed_subscriptions"][
+                    str(subscription["subscription_id"])
+                ] = {"entity_id": str(entity.entity_id), "dashboard_ids": []}
+
+            # Add permissions to user
+
+            if user_id != admin_user_id:
+                # Add permissions to user
+
+                if "permissions_granted" not in stages[user_id]:
+                    try:
+                        add_collection_permissions_to_user(
```

Comment:
We are using this func only in one place, here, probably better to simplify code and remove function `add_collection_permissions_to_user` copy code from it here.